fms/models/llama.py

```diff
@@ -531,22 +532,68 @@ def factory(**kwargs):
     _llama_factory_factory((_granite_8b_code_config)),
 )
 
+# Create all the pieces to generate adapters for different checkpoints
+_legacy_unfused_to_fused = (
+    lambda sd, ea: serialization._mlp_glu_unfused_to_fused_adapter(
+        serialization._attn_unfused_to_fused_adapter(sd, {"legacy": True})
+    )
+)
+serialization.register_adapter_step(
+    "llama", "pre0.0.6_unfused_to_fused", _legacy_unfused_to_fused
+)
+
+_unfused_to_fused = lambda sd, ea: serialization._mlp_glu_unfused_to_fused_adapter(
+    serialization._attn_unfused_to_fused_adapter(sd, {"legacy": False})
+)
+
+
+def _weight_fusion(input_sd: Mapping, extra_kwargs: Optional[Mapping] = None):
+    has_fused_weights = True
+    if extra_kwargs and "model_config" in extra_kwargs:
+        if not extra_kwargs["model_config"]["fused_weights"]:
+            has_fused_weights = False
+
+    new_sd = input_sd
+    if has_fused_weights:
+        new_sd = _unfused_to_fused(new_sd, extra_kwargs)
```

Comment:
this seems like the opposite of what was happening in `gpt_bigcode`, was that intended?

Reply:
yes